src/validation.cpp

```diff
@@ -2206,13 +2206,42 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
     return true;
 }
 
+CoinsCacheSizeState CChainState::GetCoinsCacheSizeState(
+    const CTxMemPool& tx_pool,
+    size_t max_coins_cache_size_bytes,
+    int64_t max_mempool_size_bytes)
+{
+    if (max_coins_cache_size_bytes == -1) {
```

Comment:
These magic `-1` values do not seem like the best thing. They aren't explicitly documented, assign a negative number to an unsigned type, add a runtime condition for something that is known at compile time, and cause the function to be linked against global variables that it doesn't always need.

Would suggest using an overload instead of default values:

```c++
CoinsCacheSizeState GetCoinsCacheSizeState(const CTxMemPool& tx_pool, size_t max_coins_cache_size_bytes, int64_t max_mempool_size_bytes);
CoinsCacheSizeState GetCoinsCacheSizeState(const CTxMemPool& tx_pool) {
    return GetCoinsCacheSizeState(tx_pool, ::nCoinCacheUsage, gArgs.GetArg("-maxmempool", DEFAULT_MAX_MEMPOOL_SIZE) * 1000000);
}
```

Reply:
Good points, will fix.